megamek/src/megamek/utilities/ai/UtilityPathRankerCostFunction.java

```diff
@@ -0,0 +1,1204 @@
+                        coverScore += 0.2;
+                    }
+                }
+            }
+        } else {
+            // Open terrain may be less desirable.
+            coverScore -= 0.05;
+        }
+        // Return an average bonus (or simply the sum—adjust as desired)
+        return coverScore;
+    }
+
+    private double calculateExpectedDamage(UnitAction action, Map<Integer, UnitState> states,
+                                           BehaviorParameters params) {
+        Coords finalPos = action.finalPosition();
+        double damageSum = nonAlliedUnits(action, states).stream()
+            .filter(e -> e.position().distance(finalPos) <= e.maxRange())
+            .mapToDouble(e -> getDamageAtRange(finalPos.distance(e.position()), e))
+            .sum();
+
+        return clamp01(1 - params.p12() * damageSum/100);
+    }
+
+    private double calculateAdvancedCoverage(UnitAction action, Map<Integer, UnitState> states,
+                                             BehaviorParameters params) {
+        long coveringAllies = alliedUnits(action, states).stream()
+            .filter(a -> a.position().distance(action.finalPosition()) <= a.maxRange() * 0.6)
+            .filter(a -> hasLineOfSight(a.position(), action.finalPosition()))
+            .count();
+
+        double baseCoverage = 0.8 + (coveringAllies * params.p13());
+        double densityBonus = calculateCoverDensity(action.finalPosition(), 3);
+
+        return clamp01(baseCoverage + params.p21() * densityBonus);
+    }
+
+    private double calculateImmediateEnvironmentMod(UnitAction action,
+                                                    BehaviorParameters params) {
+        Hex finalHex = board.getHex(action.finalPosition());
+        if (finalHex == null) {
+            return 1;
+        }
+        double coverScore = 0;
+        int currentHeight = boardTerrainLevels[action.toY() * board.getWidth() + action.toX()];
+        List<Coords> hexesAround = finalHex.getCoords().allAdjacent();
+        for (var targetPosition : hexesAround) {
+            Hex targetHex = board.getHex(targetPosition);
+            if (targetHex == null) continue;
+
+            coverScore += calculateHeightCover(action.finalPosition(), currentHeight);
+            // Water depth analysis
+            coverScore += getWaterCoverScore(params, targetHex);
+
+            // Surrounding terrain analysis
+            coverScore += calculateTerrainCover(action.finalPosition());
+
+            // Building cover
+            coverScore += calculateBuildingCovert(params, targetPosition);
+        }
+        return 1 + coverScore;
+    }
+
+    private static double getWaterCoverScore(BehaviorParameters params, Hex targetHex) {
+        if (targetHex.hasDepth1WaterOrDeeper()) {
+            return 0.3 * params.p15();
+        }
+        return 0.0;
+    }
+
+    private double calculateBuildingCovert(BehaviorParameters params, Coords targetPosition) {
```

Comment:
Should this be `calculateBuildingCover`?  Or is it "Covert" for being hidden?

Reply:
Should be cover